lib/computer_player.rb

```diff
@@ -0,0 +1,19 @@
+class ComputerPlayer
+  attr_accessor :computer_choices, :grid
+
+  def initialize(grid)
+    @computer_choices = []
```

Comment:
These `computer_choices` was hard to follow. From what I can guess having taken a couple passes, the grid returns all valid choices and then it is stored.

While this class only has `take_turn` as a public method so the risk of stale data is not present. It is weird that the class holds on to this data between turns.

If we think about the state as being essential to the object, because objects are combinations of state with methods, then that it exists in this out of sync world where the available computer choices become wrong as soon as the other player takes a turn ... it feels wrong.

Could we just ask the grid for a single random option? If you want the computer to have the option to make its own choice without the grid being involved. What if this computer player had its own grid, independent from the shared game grid. In that way it could ask for options that are already bounded and valid.

This would mean that the class would become closer to the input player class because it would loop to ask if a coordinate was valid.